.github/workflows/build_and_test.yaml

```diff
@@ -16,54 +37,110 @@ on:
 
 jobs:
   build:
-    name: Build, Lint and Test
+    name: Build and Test
     runs-on: ${{ inputs.runner || 'ubuntu-22.04' }}
+    env:
+      CACHE_PATH: cache
+      CI_REPO: EVerest/everest-core
+      CI_REPO_DIR: ci-repo
+      SOURCE_DIR: source
     steps:
       - name: Format branch name for cache key
         run: |
           BRANCH_NAME_FOR_CACHE="${GITHUB_REF_NAME//-/_}"
           echo "branch_name_for_cache=${BRANCH_NAME_FOR_CACHE}" >> "$GITHUB_ENV"
-      - name: Setup cache
-        uses: actions/cache@v3
+      - name: Restore cache
+        uses: actions/cache/restore@v3
+        id: restore-cache
         with:
-          path: cache
+          path: ${{ env.CACHE_PATH }}
           key: compile-${{ env.branch_name_for_cache }}-${{ github.sha }}
           restore-keys: |
             compile-${{ env.branch_name_for_cache }}-
             compile-
-      - name: Checkout everest-core
-        uses: actions/checkout@v3
+      - name: Checkout `${{ env.CI_REPO }}` to `${{ env.CI_REPO_DIR }}/` (for reusable workflow mode, see https://stackoverflow.com/a/74123003/1168315)
+        uses: actions/checkout@v4
         with:
-          path: source
-      - name: Run clang-format
-        uses: everest/everest-ci/github-actions/run-clang-format@v1.0.0
+          path: ${{ env.CI_REPO_DIR }}
+          repository: ${{ env.CI_REPO }}
+          ref: CI/reusable-workflow  # TODO get reference in general, see https://stackoverflow.com/questions/74784735 resp. https://github.com/actions/toolkit/issues/1264
```

Comment:
@andistorm Do you know a solution for this? Otherwise, I would just set it to `main` for now.

Reply:
I would suggest to not checkout the ci repo in the workflow, but to use custom github actions. There are multiple options for github actions: `composite actions` that are similiar to `reusable workflows`, `Javascript actions` and `Docker container actions`. The last one would be an option to include additional filesto the action.

I could also imagine a solution with having the bash scripts in each repo with "custom" commands and cmake flags..